ddtrace/tracer/payload.go

```diff
@@ -45,36 +44,38 @@ var _ io.Reader = (*payload)(nil)
 func newPayload() *payload {
 	p := &payload{
 		header: make([]byte, 8),
+		buf:    new(bytes.Buffer),
```

Comment:
What I meant was to completely remove this line, and to not make `bytes.Buffer` a pointer in the structure. I found that's the way that `bytes.Buffer` is generally used. We can just use ampersand `&` to reference it everywhere else.